### DISCUSSION TITLE
Adding new network checks to help with troubleshooting

### DISCUSSION THREAD

**Discussion by tteck** [tteck/Proxmox#468] [Announcements]
### Changed

- **All LXC's** 
  - Add Internet & DNS Check

```
   __  ____                __       
  / / / / /_v4__  ______  / /___  __
 / / / / __ \/ / / / __ \/ __/ / / /
/ /_/ / /_/ / /_/ / / / / /_/ /_/ / 
\____/_.___/\__,_/_/ /_/\__/\__,_/  

Using Advanced Settings
Using Ubuntu Version: 22.04
Using Container Type: 1
Using Root Password: Automatic Login
Using Container ID: 102
Using Hostname: ubuntu
Using Disk Size: 2
Allocated Cores: 1
Allocated RAM: 512
Using Bridge: vmbr0
Using IP Address: 192.168.86.252/24
Using Gateway IP Address: 192.168.86.1
Using MAC Address: Default
Using Vlan: Default
Creating a Ubuntu LXC using the above advanced settings
 ✓ Using local for Template Storage.
 ✓ Using local-lvm for Container Storage.
 ✓ Updated LXC Template List
 ✓ Downloaded LXC Template
 ✓ LXC Container 102 was successfully created.
 ✓ Started LXC Container
 ✓ Set up Container OS
 ✓ Network Connected: 192.168.86.252 
➜ ✓ Internet Connected
➜ ✓ DNS Resolved github.com to 140.82.113.4
 ✓ Updated Container OS
 ✓ Installed Dependencies
 ✓ Customized Container
 ✓ Cleaned
 ✓ Completed Successfully!
```

**Comment by Schiiiiins** [comment]
It always says, that I'm not connected to the internet and the script breaks at that point. But the container is already running and when I go to it's console I can update manually, ping IPs and hostnames:

<img width="665" alt="Screenshot 2022-10-14 at 22 31 58" src="https://user-images.githubusercontent.com/5199549/195938680-dcbc8445-aef8-41f3-accf-6da355efa9a9.png">

**Reply by tteck** [reply]
add option to continue upon Internet NOT Connected [#727](https://github.com/tteck/Proxmox/pull/727)

**Comment by DDomnick** [comment]
Could this new network check perhaps be the reason the LXC scripts are failing with No Network errors on two separate ProxMox installs but working on a third? Fails with multiple scripts tried (MQTT shown as example below).  The two failing are in a 10.10.0.0/16 subnet yet both have full internet connections after the failure.  The working site is in a 192.168.1.0/24 subnet.

`    __  _______  ____________
   /  |/  / __ \/_  __/_  __/
  / /|_/ / / / / / /   / /   
 / /  / / /_/ / / /   / /    
/_/  /_/\___\_\/_/ v4/_/     

Using Default Settings
Using Container Type: Unprivileged NO DEVICE PASSTHROUGH
Using Root Password: Automatic Login
Using Container ID: 100
Using Hostname: mqtt
Using Disk Size: 2GB
Allocated Cores 1
Allocated Ram 512
Using Bridge: vmbr0
Using Static IP Address: dhcp
Using Gateway Address: Default
Using MAC Address: Default
Using VLAN Tag: Default
Creating a MQTT LXC using the above default settings
 ✓ Using local for Template Storage.
 ✓ Using local-lvm for Container Storage.
 ✓ Updated LXC Template List
 ✓ LXC Container 100 was successfully created.
 ✓ Started LXC Container
 - Setting up Container OS ...✗ No Network! ✗ No Network! ✗ No Network! ✗ No Network! ✗ No Network! ✗ No Network! ✗ No Network! ✗ No Network! ✗ No Network! ✗ No Network! ✗ No Network After 10 Tries


root@proxmox-h77mitx:~# ip address
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: enp2s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast master vmbr0 state UP group default qlen 1000
    link/ether bc:5f:f4:43:af:f5 brd ff:ff:ff:ff:ff:ff
3: vmbr0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether bc:5f:f4:43:af:f5 brd ff:ff:ff:ff:ff:ff
    inet 10.10.0.201/16 scope global vmbr0
       valid_lft forever preferred_lft forever
    inet6 fe80::be5f:f4ff:fe43:aff5/64 scope link 
       valid_lft forever preferred_lft forever
4: veth100i0@if2: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master vmbr0 state UP group default qlen 1000
    link/ether fe:38:66:1f:9d:2f brd ff:ff:ff:ff:ff:ff link-netnsid 0


root@proxmox-h77mitx:~ ping google.com
PING google.com (142.251.163.102) 56(84) bytes of data.
64 bytes from wv-in-f102.1e100.net (142.251.163.102): icmp_seq=1 ttl=102 time=25.1 ms
64 bytes from wv-in-f102.1e100.net (142.251.163.102): icmp_seq=2 ttl=102 time=25.1 ms
64 bytes from wv-in-f102.1e100.net (142.251.163.102): icmp_seq=3 ttl=102 time=21.6 ms
64 bytes from wv-in-f102.1e100.net (142.251.163.102): icmp_seq=4 ttl=102 time=19.5 ms
^C
--- google.com ping statistics ---
4 packets transmitted, 4 received, 0% packet loss, time 3004ms
rtt min/avg/max/mdev = 19.527/22.824/25.126/2.394 ms
root@proxmox-h77mitx:~`

**Reply by tteck** [reply]
This is the oldest of the checks "Network Check" all it's doing is a `hostname -I` looking for a valid IP address. 
This is the LXC failing to receive a DHCP assigned IP address.

**Reply by DDomnick** [reply]
Found DHCP server rejecting 'randomized' mac prefixes, thanks for the lead.  Coffee on me tomorrow, thanks for the scripts.
